prov/gni/test/rdm_dgram_rma.c

```diff
@@ -1604,6 +1604,164 @@ Test(dgram_rma_1dom, readmsg)
 	xfer_for_each_size(do_readmsg, 8, BUF_SZ);
 }
 
+void do_readmsg_more(int len, void *s, void *t, int len2, void *s2, void *t2)
+{
+	int ret;
+	ssize_t sz;
+	struct fi_cq_tagged_entry cqe = { (void *) -1, UINT_MAX, UINT_MAX,
+					  (void *) -1, UINT_MAX, UINT_MAX };
+	struct iovec iov, iov2;
+	struct fi_msg_rma msg, msg2;
+	struct fi_rma_iov rma_iov, rma_iov2;
+	uint64_t w[2] = {0}, r[2] = {0}, w_e[2] = {0}, r_e[2] = {0};
+
+	iov.iov_base = s;
+	iov.iov_len = len;
+
+	rma_iov.addr = (uint64_t)t;
+	rma_iov.len = len;
+	rma_iov.key = mr_key[1];
+
+	msg.msg_iov = &iov;
+	msg.desc = (void **)loc_mr;
+	msg.iov_count = 1;
+	msg.addr = gni_addr[1];
+	msg.rma_iov = &rma_iov;
+	msg.rma_iov_count = 1;
+	msg.context = t;
+	msg.data = (uint64_t)t;
+
+	iov2.iov_base = s2;
+	iov2.iov_len = len2;
+
+	rma_iov2.addr = (uint64_t)t2;
+	rma_iov2.len = len2;
+	rma_iov2.key = mr_key2[1];
+
+	msg2.msg_iov = &iov2;
+	msg2.desc = (void **)loc_mr2;
+	msg2.iov_count = 1;
+	msg2.addr = gni_addr[1];
+	msg2.rma_iov = &rma_iov2;
+	msg2.rma_iov_count = 1;
+	msg2.context = t2;
+	msg2.data = (uint64_t)t2;
+
+
+	init_data(t, len, 0xef);
+	init_data(t2, len2, 0xff);
+	init_data(s, len, 0);
+	init_data(s2, len2, 0);
+
+	sz = fi_readmsg(ep[0], &msg, FI_MORE);
+	cr_assert_eq(sz, 0);
+	sz = fi_readmsg(ep[0], &msg2, 0);
+	cr_assert_eq(sz, 0);
+
+	while ((ret = fi_cq_read(send_cq[0], &cqe, 1)) == -FI_EAGAIN) {
+		pthread_yield();
+	}
+
+	cr_assert_eq(ret, 1);
+	rdm_rma_check_tcqe(&cqe, t, FI_RMA | FI_READ, 0);
+
+	while ((ret = fi_cq_read(send_cq[0], &cqe, 1)) == -FI_EAGAIN) {
+		pthread_yield();
+	}
+
+	cr_assert_eq(ret, 1);
+	rdm_rma_check_tcqe(&cqe, t2, FI_RMA | FI_READ, 0);
+
+	r[0] = 2;
+	rdm_rma_check_cntrs(w, r, w_e, r_e);
+
+	dbg_printf("got write context event!\n");
+
+	cr_assert(check_data(s, t, len), "Data mismatch1");
+	cr_assert(check_data(s2, t2, len2), "Data mismatch2");
+}
+
+void do_read_alignment_more(void)
```

Comment:
Is it possible to replace the constants below with values defined in the provider?  That way, if the defaults change, we are still testing we we think we are testing.